### PR TITLE
HPCC4J-630 Ensure Filespray path constructed correctly

### PR DESCRIPTION
**Pull request by rpastrana**
- Add util method to append paths
- Add junit test for new methods
- Ensure wsfs methods use new method

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [x] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 1]
```diff
@@ -1005,7 +1005,7 @@ public ProgressResponseWrapper sprayVariable(DelimitedDataOptions options, DropZ
 
         SprayVariable request = new SprayVariable();
         request.setSourceIP(targetDropZone.getNetAddress());
-        request.setSourcePath(Utils.ensureTrailingPathSlash(targetDropZone.getPath()) + sourceFileName);
+        request.setSourcePath(Utils.ensureTrailingPathSlash(Utils.appendPathSections(targetDropZone.getPath(),sourceFileName)));
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java:1008]: Should there be a Utils function that builds the path and ensures there is trailing path slash?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 2]
```diff
@@ -1317,7 +1317,7 @@ public ProgressResponseWrapper sprayFixed(DropZoneWrapper targetDropZone, String
         request.setDestGroup(destGroup);
         request.setSourceRecordSize(recordSize);
         request.setSourceIP(targetDropZone.getNetAddress());
-        request.setSourcePath(Utils.ensureTrailingPathSlash(targetDropZone.getPath()) + sourceFileName);
+        request.setSourcePath(Utils.ensureTrailingPathSlash(Utils.appendPathSections(targetDropZone.getPath(), sourceFileName)));
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java:1320]: What are the valid path schemas? Is it just Linux or are Windows paths valid?

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/Utils.java [hunk 3]
```diff
@@ -1114,6 +1114,121 @@ public static String ensureTrailingPathSlash(String path, char slash)
...
+            return "";
+
+        try
+        {
+            char pathSep = inferPathSeperatorType(prefixPath.length() != 0 ? prefixPath : postfixPath);
+            return appendPathSections(prefixPath, pathSep, postfixPath);
+        }
+        catch (Exception e)
+        {
+            throw new Exception("Could not append path sections, ensure original path sections are valid contain path seperator");
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/Utils.java:1173]: Typo here missing an and

**Comment by rpastrana** [comment]
@jpmcmu please take a look again. You're right, there was no longer a need to ensure the trailing path on the properly appended path

**Review by jpmcmu — APPROVED** [review]
@rpastrana looks good to me

**Comment by rpastrana** [comment]
@jchambers-ln this pr is ready to merge, existing tests passed. We can hold up the merge, If you'd like to verify fix

**Comment by jchambers-ln** [comment]
I've verified this should fix the issue; I didn't go sniffing all the edge cases, but it solves the issue we were seeing with alpha dev not being able to spray files.
